src/algorithms/synthesis/encoding.cpp

```diff
@@ -126,21 +101,57 @@ namespace syrec {
             output = encoding[output];
         }
 
-        return {encoding, additionalLines};
+        return encoding;
     }
 
-    auto augmentWithConstants(TruthTable& tt, std::size_t const& nBits, bool dc) -> void {
-        const auto ancillaBits = nBits - tt.nOutputs();
+    auto augmentWithConstants(TruthTable& tt, std::size_t const& nBits, bool appendZero) -> void {
+        const auto requiredOutConstants = nBits - tt.nOutputs();
+        const auto requiredInConstants  = nBits - tt.nInputs();
 
         for (auto& [input, output]: tt) {
-            // add necessary constant inputs to the outputs based on the total number of bits (nBits).
-            if (!dc) {
-                for (auto i = 0U; i < ancillaBits; i++) {
-                    output.insertZero();
+            const auto oldGarbageVec         = tt.getGarbage();
+            const auto currentGarbageVecSize = oldGarbageVec.size();
+
+            if (appendZero) {
+                // based on the requiredOutConstants, zeros are appended to the outputs.
+                for (auto i = 0U; i < requiredOutConstants; i++) {
+                    // based on the requiredOutConstants, zeros are appended to the outputs.
+                    output.emplace_back(false);
+
+                    if (currentGarbageVecSize != nBits) {
+                        // reset and resize the garbage.
+                        tt.resetGarbage();
+                        tt.resizeGarbage(currentGarbageVecSize + 1);
+
+                        for (auto cg = 0U; cg < currentGarbageVecSize; cg++) {
+                            if (oldGarbageVec[cg]) {
+                                // storing the old garbage bits in proper order.
+                                tt.setGarbage(cg + 1);
+                            }
+                        }
+                        // setting the newly added garbage bits.
+                        tt.setGarbage(0);
+                    }
                 }
+
             } else {
-                output.resize(nBits);
+                for (auto i = 0U; i < requiredOutConstants; i++) {
+                    // based on the requiredOutConstants, zeros are inserted to the outputs.
+                    output.insertZero();
+                    if (currentGarbageVecSize != nBits) {
+                        // reset and resize the garbage.
+                        tt.resetGarbage();
+                        tt.resizeGarbage(currentGarbageVecSize + 1U);
+                        for (auto cg = 0U; cg < currentGarbageVecSize; cg++) {
+                            // storing the old garbage bits in proper order.
+                            if (oldGarbageVec[cg]) {
+                                tt.setGarbage(cg);
+                            }
+                        }
+                    }
```

Comment:
Shouldn't this part of the code also declare the newly added bit as garbage? at least in the `appendZero` branch that is the case.

Reply:
There is no garbage bit being added in this case. A new bit is being added so that the primary outputs can be stored. Hence, it is not set to garbage.

Reply:
Makes sense. Was just wondering. Feel free to resolve.